zeebe/backup/src/main/java/io/camunda/zeebe/backup/metrics/BackupManagerMetrics.java

```diff
@@ -7,102 +7,112 @@
  */
 package io.camunda.zeebe.backup.metrics;
 
-import io.prometheus.client.Counter;
-import io.prometheus.client.Gauge;
-import io.prometheus.client.Histogram;
-import io.prometheus.client.Histogram.Timer;
+import static io.camunda.zeebe.backup.metrics.BackupManagerMetricsDoc.*;
+
+import io.camunda.zeebe.util.CloseableSilently;
+import io.camunda.zeebe.util.collection.Table;
+import io.camunda.zeebe.util.micrometer.MicrometerUtil;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Timer;
+import java.util.EnumMap;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
 
 public class BackupManagerMetrics {
-  private static final String NAMESPACE = "zeebe";
-  private static final String LABEL_NAME_PARTITION = "partition";
-  private static final String LABEL_NAME_OPERATION = "operation";
-  private static final String LABEL_NAME_RESULT = "result";
-
-  private static final String FAILED = "failed";
-  private static final String COMPLETED = "completed";
-  private static final String TAKE_OPERATION = "take";
-  private static final String STATUS_OPERATION = "status";
-  private static final String LIST_OPERATION = "list";
-  private static final String DELETE_OPERATION = "delete";
-
-  private static final Counter TOTAL_OPERATIONS =
-      Counter.build()
-          .namespace(NAMESPACE)
-          .name("backup_operations_total")
-          .help("Total number of backup operations")
-          .labelNames(LABEL_NAME_PARTITION, LABEL_NAME_OPERATION, LABEL_NAME_RESULT)
-          .register();
-  private static final Gauge OPERATIONS_IN_PROGRESS =
-      Gauge.build()
-          .namespace(NAMESPACE)
-          .name("backup_operations_in_progress")
-          .help("Number of backup operations that are in progress")
-          .labelNames(LABEL_NAME_PARTITION, LABEL_NAME_OPERATION)
-          .register();
-
-  private static final Histogram BACKUP_OPERATION_LATENCY =
-      Histogram.build()
-          .namespace(NAMESPACE)
-          .name("backup_operations_latency")
-          .help("Latency of backup operations")
-          .labelNames(LABEL_NAME_PARTITION, LABEL_NAME_OPERATION)
-          .buckets(0.01, 0.1, 1, 10, 60, 5 * 60)
-          .register();
-
-  private final String partitionId;
-
-  public BackupManagerMetrics(final int partitionId) {
-    this.partitionId = String.valueOf(partitionId);
+
+  private final MeterRegistry registry;
+  private final Table<OperationType, OperationResult, Counter> totalOperations = Table.simple();
```

Comment:
:wrench: If `OperationType` and `OperationResult` are both enums, then let's use `Table.ofEnum`.